src/main/java/com/shapesecurity/salvation/data/Policy.java

```diff
@@ -268,6 +268,9 @@ public void intersectDirective(@Nonnull Directive<? extends DirectiveValue> dire
             this.expandDefaultSrc();
             this.resolveSelf();
             this.optimise();
+        } else {
+            throw new IllegalArgumentException(
```

Comment:
This shouldn't be throwing because it should be allowed according to spec. The warning should be handled entirely within the parser.